src/frontend/src/optimizer/plan_node/utils.rs

```diff
@@ -382,9 +383,23 @@ pub fn infer_kv_log_store_table_catalog_inner(
 
     let read_prefix_len_hint = table_catalog_builder.get_current_pk_len();
 
-    let payload_indices = table_catalog_builder.extend_columns(columns);
+    if columns.len() != input.schema().fields().len()
+        || columns
+            .iter()
+            .zip_eq_fast(input.schema().fields())
+            .any(|(c, f)| *c.data_type() != f.data_type())
+    {
+        tracing::warn!(
+            "sink schema different with upstream schema: sink columns: {:?}, input schema: {:?}.",
+            columns,
+            input.schema()
+        );
+    }
+    for field in input.schema().fields() {
+        let indice = table_catalog_builder.add_column(field);
+        value_indices.push(indice);
+    }
```

Comment:
cc @wenym1 PTAL

Reply:
Changes in this part LGTM. Maybe we don't need to pass the sink columns to this method, since it's only for check and warning log.